client/src/services/dataSource.js

```diff
@@ -0,0 +1,22 @@
+import api from './api';
+
+const DataSource = {
+  data: {
+    loggedUser: {},
+    users: [],
+    channels: [],
+  },
+  subscribers: [],
```

Comment:
also, I'm not sure the `data` should actually be here. My understanding is that you will use socket.io for sending and receiving messages. When a message is received, you just pass it to the subscribers and they do with it whatever they want. But I might be missing something - what is `data` going to be used for?

Reply:
well since we have multiple channels and a lot of users
and a lot of 'people' that the user can talk with we are gonna overhaul some of the UI to use this data source as the only way to connect to the server
might be a bit inefficient bc sometimes components who don't even need that data will get notified
but again:
their unsubscribe on component did mount is something that avoids unnecessary notifications
am i right?